dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -681,7 +669,7 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                 destination,
                 ..
             } => {
-                let mut arg_local = mir_loc::Local { index: 1 };
+                let mut callee_arg: Place = Local::from_u32(1).into();
```

Comment:
```suggestion
                let mut callee_arg: Place = Local::new(1).into();
```

Requires `use rustc_index::vec::Idx` (separate suggestion).

I think it makes more sense to use the `Idx` methods, as we're using this as an index and we don't have to hardcode the `u32` here.  The `Idx::increment_by` method makes more sense than the previous code, so I think it'd be better if we stick to the `Idx` methods for `callee_arg`.